[Camel K] Integrations failing at startup

## This is a...

<pre><code>
[ ] Feature request
[ ] Regression (a behavior that used to work and stopped working in a new release)
[x] Bug report  <!-- Please search GitHub for a similar issue or PR before submitting -->
[ ] Documentation issue or request
</code></pre>

## Description

Any integration running with `camel k` runtime is failing due to the following error:
```
 [ERROR] Failed to execute goal on project camel-k-integration: Could not resolve dependencies for project org.apache.camel.k.integration:camel-k-integration:jar:1.0.0-RC2: Could not find artifact io.atlasmap:camel3-atlasmap:jar:1.42.10 in central (https://repo.maven.apache.org/maven2) -> [Help 1]
```
I also noticed there is a misalignment with the jackson libraries, so we need to fix both to have integrations running.

After fixing the issue described, we have another failure:
```
I> No access restrictor found, access to any MBean is allowed
Jolokia: Agent started with URL https://10.128.0.73:8778/jolokia/
2020-06-22 10:40:00.744 [32mINFO [m [main] ApplicationRuntime - Add listener: org.apache.camel.k.listener.ContextConfigurer@59e2d8e3
2020-06-22 10:40:00.755 [32mINFO [m [main] ApplicationRuntime - Add listener: org.apache.camel.k.listener.RoutesConfigurer@48c76607
2020-06-22 10:40:00.756 [32mINFO [m [main] ApplicationRuntime - Add listener: org.apache.camel.k.listener.RoutesDumper@6b6776cb
2020-06-22 10:40:00.758 [32mINFO [m [main] ApplicationRuntime - Add listener: org.apache.camel.k.listener.PropertiesFunctionsConfigurer@53b7f657
2020-06-22 10:40:00.784 [32mINFO [m [main] ApplicationRuntime - Listener org.apache.camel.k.listener.PropertiesFunctionsConfigurer@53b7f657 executed in phase Starting
2020-06-22 10:40:00.793 [32mINFO [m [main] RuntimeSupport - Looking up loader for language: syndesis
2020-06-22 10:40:00.796 [32mINFO [m [main] RuntimeSupport - Found loader io.syndesis.integration.runtime.camelk.IntegrationRouteLoader@45a4b042 for language syndesis from service definition
2020-06-22 10:40:00.797 [32mINFO [m [main] IntegrationRouteLoader - Loading ActivityTracker from Camel RuntimeRegistry.
2020-06-22 10:40:00.797 [32mINFO [m [main] IntegrationRouteLoader - ActivityTracker not provided or not found in Camel RuntimeRegistry, using new instance of ActivityTracker.SysOut() .
2020-06-22 10:40:00.797 [32mINFO [m [main] IntegrationRouteLoader - Loading IntegrationStepHandlers with ServiceLoader.
2020-06-22 10:40:00.936 [32mINFO [m [main] IntegrationRouteLoader - 13 IntegrationStepHandlers loaded.
2020-06-22 10:40:00.936 [32mINFO [m [main] IntegrationRouteLoader - Loading tracer from Camel RuntimeRegistry.
2020-06-22 10:40:00.936 [32mINFO [m [main] IntegrationRouteLoader - No tracer found.
2020-06-22 10:40:00.945 [32mINFO [m [main] RoutesConfigurer - Loading routes from: file:/etc/camel/sources/i-source-000/i-maqzkaslnnw-y4jzyslz?language=syndesis
2020-06-22 10:40:00.945 [32mINFO [m [main] ApplicationRuntime - Listener org.apache.camel.k.listener.RoutesConfigurer@48c76607 executed in phase ConfigureRoutes
2020-06-22 10:40:00.949 [32mINFO [m [main] BaseMainSupport - Using properties from: file:/etc/camel/conf/application.properties,file:/etc/camel/conf.d/_secrets/i-sadsa23/application.properties,file:/etc/camel/conf.d/_secrets/i-sadsa23/..2020_06_22_10_39_35.472457508/application.properties
2020-06-22 10:40:01.164 [32mINFO [m [main] BaseMainSupport - Auto-configuration summary:
2020-06-22 10:40:01.165 [32mINFO [m [main] BaseMainSupport - 	camel.context.streamcaching=true
Exception in thread "main" java.lang.NoClassDefFoundError: com/fasterxml/jackson/databind/ser/std/ToStringSerializerBase
	at java.lang.ClassLoader.defineClass1(Native Method)
	at java.lang.ClassLoader.defineClass(Unknown Source)
	at java.security.SecureClassLoader.defineClass(Unknown Source)
	at java.net.URLClassLoader.defineClass(Unknown Source)
	at java.net.URLClassLoader.access$100(Unknown Source)
	at java.net.URLClassLoader$1.run(Unknown Source)
	at java.net.URLClassLoader$1.run(Unknown Source)
	at java.security.AccessController.doPrivileged(Native Method)
	at java.net.URLClassLoader.findClass(Unknown Source)
	at java.lang.ClassLoader.loadClass(Unknown Source)
	at sun.misc.Launcher$AppClassLoader.loadClass(Unknown Source)
	at java.lang.ClassLoader.loadClass(Unknown Source)
	at io.syndesis.common.util.json.JsonUtils.<clinit>(JsonUtils.java:54)
	at io.syndesis.integration.runtime.IntegrationRouteBuilder.loadIntegration(IntegrationRouteBuilder.java:115)
	at io.syndesis.integration.runtime.IntegrationRouteBuilder.configure(IntegrationRouteBuilder.java:143)
	at org.apache.camel.builder.RouteBuilder.checkInitialized(RouteBuilder.java:471)
	at org.apache.camel.builder.RouteBuilder.configureRoutes(RouteBuilder.java:438)
	at org.apache.camel.builder.RouteBuilder.addRoutesToCamelContext(RouteBuilder.java:418)
	at org.apache.camel.impl.engine.AbstractCamelContext.lambda$addRoutes$0(AbstractCamelContext.java:1119)
	at org.apache.camel.impl.engine.AbstractCamelContext.doWithDefinedClassLoader(AbstractCamelContext.java:2460)
	at org.apache.camel.impl.engine.AbstractCamelContext.addRoutes(AbstractCamelContext.java:1119)
	at org.apache.camel.main.RoutesConfigurer.configureRoutes(RoutesConfigurer.java:84)
	at org.apache.camel.main.BaseMainSupport.postProcessCamelContext(BaseMainSupport.java:570)
	at org.apache.camel.main.BaseMainSupport.initCamelContext(BaseMainSupport.java:422)
	at org.apache.camel.k.main.ApplicationRuntime$MainAdapter.doInit(ApplicationRuntime.java:180)
	at org.apache.camel.support.service.ServiceSupport.init(ServiceSupport.java:80)
	at org.apache.camel.support.service.ServiceSupport.start(ServiceSupport.java:108)
	at org.apache.camel.main.MainSupport.run(MainSupport.java:77)
	at org.apache.camel.k.main.ApplicationRuntime.run(ApplicationRuntime.java:76)
	at org.apache.camel.k.main.Application.main(Application.java:42)
Caused by: java.lang.ClassNotFoundException: com.fasterxml.jackson.databind.ser.std.ToStringSerializerBase
	at java.net.URLClassLoader.findClass(Unknown Source)
	at java.lang.ClassLoader.loadClass(Unknown Source)
	at sun.misc.Launcher$AppClassLoader.loadClass(Unknown Source)
	at java.lang.ClassLoader.loadClass(Unknown Source)
	... 30 more
2020-06-22 10:40:01.211 [32mINFO [m [Camel Thread #0 - CamelHangupInterceptor] MainSupport$HangupInterceptor - Received hang up - stopping the main instance.
2020-06-22 10:40:01.212 [32mINFO [m [Camel Thread #0 - CamelHangupInterceptor] DefaultCamelContext - Apache Camel 3.0.1 (CamelContext: camel-k) is shutting down
2020-06-22 10:40:01.220 [32mINFO [m [Camel Thread #0 - CamelHangupInterceptor] DefaultCamelContext - Apache Camel 3.0.1 (CamelContext: camel-k) uptime 
2020-06-22 10:40:01.222 [32mINFO [m [Camel Thread #0 - CamelHangupInterceptor] DefaultCamelContext - Apache Camel 3.0.1 (CamelContext: camel-k) is shutdown in 0.008 seconds

```

--- a/app/integration/bom-camel-k/pom.xml
+++ b/app/integration/bom-camel-k/pom.xml
@@ -28,10 +28,8 @@
 
   <properties>
     <camel.version>3.1.0</camel.version>
-    <atlasmap.version>1.42.10</atlasmap.version>
-    <jackson.version>2.11.0</jackson.version>
-    <jackson.databind.version>2.9.9</jackson.databind.version>
-    <jackson-module-jaxb-annotations.version>2.9.9</jackson-module-jaxb-annotations.version>
+    <atlasmap.version>2.0.4</atlasmap.version>
+    <jackson.version>2.10.2</jackson.version>
     <validation-api.version>2.0.2</validation-api.version>
   </properties>
 
@@ -95,7 +93,7 @@
       <dependency>
         <groupId>com.fasterxml.jackson.core</groupId>
         <artifactId>jackson-databind</artifactId>
-        <version>${jackson.databind.version}</version>
+        <version>${jackson.version}</version>
       </dependency>
       <dependency>
         <groupId>com.fasterxml.jackson.core</groupId>

